src/scipp/core/binning.py

```diff
@@ -488,8 +488,21 @@ def _make_groups(x, arg):
     _require_coord(arg, coord)
     if coord.bins is not None:
         coord = coord.copy().bins.constituents['data']
+    if coord.dtype in (_cpp.DType.int32, _cpp.DType.int64):
+        min_ = coord.min().value
+        max_ = coord.max().value
+        values = coord.values
+        pivot = len(values) // 100
+        unique = np.unique(values[:pivot])
+        if len(unique) != max_ - min_ + 1:
+            pivot = len(values) // 10
+            unique = np.unique(values[:pivot])
```

Comment:
Did you consider something like
```.py
unique = np.unique(np.r_[unique, np.unique(values[old_pivot:new_pivot]])
```
This might add a small speedup. But it makes the pathological case of few duplicate coord values even worse.

Reply:
If we stick to the 10x scaling in `pivot` then I think it is unlikely that you can tell the speedup in practice, but the code is more complicated (and likely to be broken)?

Reply:
Ok.
We can always revisit if it becomes an issue.